streams/src/main/java/org/apache/kafka/streams/processor/SkipInvalidConsumerRecordTimestampExtractor.java

```diff
@@ -38,36 +41,28 @@
  * using this extractor effectively provides <i>ingestion-time</i> semantics.
  * <p>
  * If you need <i>processing-time</i> semantics, use {@link WallclockTimestampExtractor}.
- * <p>
- * If a record has a negative (invalid) timestamp value the timestamp is returned as-is;
- * in addition, a WARN message is logged in your application.
- * Returning the timestamp as-is results in dropping the record, i.e., the record will not be processed.
  *
- * @see ConsumerRecordTimestampExtractor
+ * @see FailingConsumerRecordTimestampExtractor
  * @see InferringConsumerRecordTimestampExtractor
  * @see WallclockTimestampExtractor
  */
-public class RobustConsumerRecordTimestampExtractor implements TimestampExtractor {
-    private static final Logger log = LoggerFactory.getLogger(RobustConsumerRecordTimestampExtractor.class);
+public class SkipInvalidConsumerRecordTimestampExtractor extends AbstractConsumerRecordTimestampExtractor {
```

Comment:
What about a shorter name like `SkipRecordsWithInvalidTimestamps`?

Also, I don't think "TimestampExtractor" _must_ be in the name of the class. The term "ConsumerRecord" is also pretty confusing IMHO to the average user because it's not a record OF a consumer etc.